### DISCUSSION TITLE
SON-OFF Dual R3 - Push Button.

### DISCUSSION THREAD

**Discussion by miazza99** [arendst/Tasmota#14362] [Q&A] [answered]
I have one question about the pushbutton function.
While pushing the GPIO00 the switch n°1 is activate.
Where is it located the GPIO for acting on the switch n° 2 ?

**Comment by sfromis** [comment]
Not sure what you're asking about. You can see the gpio config in the [template](https://templates.blakadder.com/sonoff_DUALR3.html), including the two switch inputs (S1 and S2) already configured when you apply the template.

**Reply by miazza99** [reply]
What I'm asking is the following:
- Acording to Template GPIO00 is connected to Button 1. If I push it I can activate Relay 1 (ON and OFF).
Now, if I need to connect a push button to activate Relay 2 where shall I connect it ?

I tried to double press the button with no result.

**Reply by sfromis** [reply]
That's why I mentioned the switch inputs (S1 and S2) already. You have those, and be really careful to follow the wiring diagrams.

**Reply by miazza99** [reply]
Yes, I understand. please see my last answer to arendst.

**Comment by arendst** [comment]
Try a double press on the button ;-)

**Comment by arendst** [comment, marked as answer]
If you want two buttons you need to connect them to the assigned connectors: S1 and S2. 

Refrain from connecting wires to the internals of the Dual R3 as it's all connected to life voltage.

**Reply by miazza99** [reply]
This is what I did. But I have a particular set-up for my shutter: the shutter is right now commanded with a remote (RT3 - https://www.telecommandes-toutes-marques.com/ttm_images/divers/MHOUSE-MT4-MOOVO.pdf)

I tried to connect the RT3 Output PIN 1 and 3 into S1 and S2 while the motor is connected to LOUT 1 and LOUT 2.
This seems not working and I do not know why (it is the same Phase as L in but controlled by another relay).

**Reply by barbudor** [reply]
You probably need to apply Switchmode 1 (Follow) while the default switch mode is Toggle
https://tasmota.github.io/docs/Buttons-and-Switches/#switchmode

In follow mode, the matching output will follow the switch input : ON => ON, OFF => OFF

If still not ok, please provide a detailled diagram of your wiring

**Reply by miazza99** [reply]
Yesterday I opened the box and I found C19 burned. 
I guess this is the reason why S1 and S2 are not working :(
![DUAL R3 Burnt](https://user-images.githubusercontent.com/61536702/148895963-65e0d224-2fbe-4ab5-a88b-a70ef60e20e4.jpg)

Probably the issue occurred due to a mistake I made in the first connection (by mistake I connected RT3 to the output and the shutter motor to S1 and S2).

The device is still connecting to the wifi and working as expected except S1 and S2 that are not working at all.

Any idea about the C19 value and voltage rating ? May be I can try to clean it and substitute it.

Does anyone have an electrical schematic of the device ?
